tests/e2e/e2e_egress_test.go

```diff
@@ -0,0 +1,103 @@
+package e2e
+
+import (
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Egress", func() {
+	Context("Egress", func() {
+		sourceNs := "client"
+
+		It("Allows egress traffic when enabled", func() {
+			// Install OSM
+			installOpts := td.GetOSMInstallOpts()
+			installOpts.egressEnabled = true
+			Expect(td.InstallOSM(installOpts)).To(Succeed())
+			Expect(td.WaitForPodsRunningReady(td.osmNamespace, 60*time.Second, 1)).To(Succeed())
+
+			// Create Test NS
+			Expect(td.CreateNs(sourceNs, nil)).To(Succeed())
+			Expect(td.AddNsToMesh(true, sourceNs)).To(Succeed())
+
+			// Get simple Pod definitions for the client
+			svcAccDef, podDef, svcDef := td.SimplePodApp(SimplePodAppDef{
+				name:      "client",
+				namespace: sourceNs,
+				command:   []string{"/bin/bash", "-c", "--"},
+				args:      []string{"while true; do sleep 30; done;"},
+				image:     "songrgg/alpine-debug",
+				ports:     []int{80},
+			})
+
+			_, err := td.CreateServiceAccount(sourceNs, &svcAccDef)
+			Expect(err).NotTo(HaveOccurred())
+			srcPod, err := td.CreatePod(sourceNs, podDef)
+			Expect(err).NotTo(HaveOccurred())
+			_, err = td.CreateService(sourceNs, svcDef)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Expect it to be up and running in it's receiver namespace
+			Expect(td.WaitForPodsRunningReady(sourceNs, 60*time.Second, 1)).To(Succeed())
+
+			egressURLs := []string{
+				"edition.cnn.com",
+				"github.com",
```

Comment:
We should be testing both HTTP and HTTPS traffic with egress. 
Could you document what is being tested in the Describe or Context container.

Reply:
@shashankram Can you clarify what you mean by "what is being tested?"

Reply:
With egress there are different things that can be tested - HTTP egress, HTTPS egress etc.
It would be nice to document what is being tested as a part of the Describe and Context container messages.

Reply:
Ginkgo's way would be to have  the test separated by `It()`s, so that each seaction represents a part of the test. We are not yet there in current impl, but we should file an issue to rework the tests into sections.

Reply:
You don't need an It() for every test, depends on what is being tested. Just a Describe and Context are more than enough for many tests.